### PR TITLE
Update grid-related examples

### PR DESCRIPTION
**Pull request by jouvin**
Hopefully fixed grid examples... In addition to fixed examples, it also fixes the `test-library` logic and removes clusters that have never been working...

### CODE REVIEW

**Comment by jrha** [comment]
Fix for indenter in #41.

**Comment by jouvin** [comment]
I think we have a major flaw in the current test `test-library`. The way `create-vanilla-scdb` is called what is tested is what has been already merged: there is no attempt to merge the current PR contents and so we are not testing the modifications contained on the PR. It is why we have all the errors in the current PR in this test, despite the PR is fixing these errors.

There was a `--continuous-integration` added at the time of Travis, I'm trying to see how to do something similar in GitHub actions.

**Comment by jouvin** [comment]
@jrha I think I fixed https://github.com/quattor/template-library-examples/pull/40#issuecomment-2463021963. The remaining error in `test-library` is caused by `template-library-core` missing some AII-related templates that are added by the releaser. May be we need to fix manually the repository to ensure we can pass the tests. We have a chicken&egg problem here.

File: clusters/grid/umd4/repository/config.pan [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# Repositories related to Nagios
+variable REPOSITORY_CONFIG_NAGIOS ?= null;
+variable REPOSITORY_CONFIG_NAGIOS_INCLUDE =
+if ( exists(REPOSITORY_CONFIG_NAGIOS) && is_defined(REPOSITORY_CONFIG_NAGIOS) ) {
+    REPOSITORY_CONFIG_NAGIOS;
+} else {
+    null;
+};
+include REPOSITORY_CONFIG_NAGIOS_INCLUDE;
```
Comment (jrha) [inline clusters/grid/umd4/repository/config.pan:44]: This is strangely over-engineered.
```suggestion
include if (is_defined(REPOSITORY_CONFIG_NAGIOS)) if_exists(REPOSITORY_CONFIG_NAGIOS);
```

File: clusters/grid/umd4/repository/config.pan [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# Register repository changes with yum based spma
+include 'components/spma/config';
+'/software/components/spma/register_change' =
+if ( is_defined(QUATTOR_RELEASE) && (QUATTOR_RELEASE >= '13.2') ) {
+    append('/software/repositories');
+} else {
+    SELF;
+};
```
Comment (jrha) [inline clusters/grid/umd4/repository/config.pan:53]: Surely we don't have to deal with anyone running releases older than 13.2 now?
```suggestion
'/software/components/spma/register_change' = if (is_defined(QUATTOR_RELEASE)) {
    append('/software/repositories');
} else {
    SELF;
};
```

File: clusters/misc/nagios/repository/config.pan [hunk 3]
```diff
@@ -1,54 +1,55 @@
...
 '/software/repositories' = repository_config(SITE_REPOSITORY_CONFIG);
 
 # Repositories related to grid middleware
 # Only if it can be found in the loadpath (else this is a machine not
 # running gLite like a non UI LAL server)
-include { if_exists('repository/config/grid') };
+include if_exists('repository/config/grid');
 
 # Repositories related to Nagios
 variable REPOSITORY_CONFIG_NAGIOS ?= null;
```
Comment (jrha) [inline clusters/misc/nagios/repository/config.pan:36]:
```suggestion
```

File: clusters/openstack/icehouse/repository/config.pan [hunk 4]
```diff
@@ -17,34 +17,36 @@ variable YUM_SITE_SNAPSHOT_NS ?= YUM_SNAPSHOT_NS;
...
+    'SITE_REPOSITORY_LIST = ' + to_string(SITE_REPOSITORY_LIST) + "\n" +
+    'SITE_REPOSITORY_CONFIG = ' + to_string(SITE_REPOSITORY_CONFIG) + "\n"
+);
 include 'quattor/functions/repository';
 '/software/repositories' = add_repositories(QUATTOR_REPOSITORY_LIST);
 '/software/repositories' = add_repositories(SITE_REPOSITORY_LIST);
 '/software/repositories' = repository_config(SITE_REPOSITORY_CONFIG);
 
 # Repositories related to Nagios
 variable REPOSITORY_CONFIG_NAGIOS ?= null;
```
Comment (jrha) [inline clusters/openstack/icehouse/repository/config.pan:32]:
```suggestion
```

File: clusters/openstack/icehouse/repository/config.pan [hunk 5]
```diff
@@ -17,34 +17,36 @@ variable YUM_SITE_SNAPSHOT_NS ?= YUM_SNAPSHOT_NS;
...
-variable REPOSITORY_CONFIG_NAGIOS_INCLUDE = {
-  if (exists(REPOSITORY_CONFIG_NAGIOS) && is_defined(REPOSITORY_CONFIG_NAGIOS)) {
+variable REPOSITORY_CONFIG_NAGIOS_INCLUDE =
+if ( exists(REPOSITORY_CONFIG_NAGIOS) && is_defined(REPOSITORY_CONFIG_NAGIOS) ) {
     REPOSITORY_CONFIG_NAGIOS;
```
Comment (jrha) [inline clusters/openstack/icehouse/repository/config.pan:35]:
```suggestion
```

File: clusters/openstack/icehouse/repository/config.pan [hunk 6]
```diff
@@ -17,34 +17,36 @@ variable YUM_SITE_SNAPSHOT_NS ?= YUM_SNAPSHOT_NS;
...
-  } else {
+} else {
     null;
```
Comment (jrha) [inline clusters/openstack/icehouse/repository/config.pan:37]:
```suggestion
```

File: clusters/openstack/icehouse/repository/config.pan [hunk 7]
```diff
@@ -17,34 +17,36 @@ variable YUM_SITE_SNAPSHOT_NS ?= YUM_SNAPSHOT_NS;
...
-    } else {
-        SELF;
-    };
+include 'components/spma/config';
+'/software/components/spma/register_change' =
+if ( is_defined(QUATTOR_RELEASE) && (QUATTOR_RELEASE >= '13.2') ) {
+    append('/software/repositories');
+} else {
+    SELF;
 };
```
Comment (jrha) [inline clusters/openstack/icehouse/repository/config.pan:48]:
```suggestion
'/software/components/spma/register_change' =
if (is_defined(QUATTOR_RELEASE)) {
    append('/software/repositories');
} else {
    SELF;
};
```

File: clusters/grid/umd4/repository/config.pan (hunk 1 above)
Comment (jouvin) [reply clusters/grid/umd4/repository/config.pan:44]: I was so focused on the indentation problem that I missed the simplifaction required. I'll fix it!

File: clusters/grid/umd4/repository/config.pan (hunk 2 above)
Comment (jouvin) [reply clusters/grid/umd4/repository/config.pan:53]: Good point! I have been too much focused on indent and panlint errors and did not review the code itself!

File: clusters/misc/nagios/repository/config.pan [hunk 8]
```diff
@@ -1,54 +1,55 @@
...
-    } else {
-        SELF;
-    };
+include 'components/spma/config';
+'/software/components/spma/register_change' =
+if ( is_defined(QUATTOR_RELEASE) && (QUATTOR_RELEASE >= '13.2') ) {
+    append('/software/repositories');
+} else {
+    SELF;
 };
```
[the remark this answers is not in the document]
Comment (jouvin) [reply clusters/misc/nagios/repository/config.pan]: Done

File: clusters/openstack/icehouse/repository/config.pan (hunk 7 above)
Comment (jouvin) [reply clusters/openstack/icehouse/repository/config.pan]: Done

File: clusters/misc/nagios/repository/config.pan [hunk 9]
```diff
@@ -1,54 +1,38 @@
...
-                       'SITE_REPOSITORY_LIST = ' + to_string(SITE_REPOSITORY_LIST) + "\n" +
-                       'SITE_REPOSITORY_CONFIG = ' + to_string(SITE_REPOSITORY_CONFIG) + "\n");
-include {'quattor/functions/repository'};
+variable SITE_REPOSITORY_CONFIG ?= dict();
+variable DEBUG = debug(
+    'OS_REPOSITORY_LIST = ' + to_string(OS_REPOSITORY_LIST) + "\n" +
+    'SITE_REPOSITORY_LIST = ' + to_string(SITE_REPOSITORY_LIST) + "\n" +
+    'SITE_REPOSITORY_CONFIG = ' + to_string(SITE_REPOSITORY_CONFIG) + "\n"
+);
+include'quattor/functions/repository'};
```
Comment (jrha) [inline clusters/misc/nagios/repository/config.pan:25]:
```suggestion
include 'quattor/functions/repository';
```

File: clusters/grid/umd4/repository/config.pan [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+# correctly.
+
+unique template repository/config;
+
+include'pan/functions';
+
+include 'repository/snapshot/snapshot_variables';
+
+# Repositories related to base OS and quattor client (should be first)
+include'repository/config/os';
```
Comment (jrha) [inline clusters/grid/umd4/repository/config.pan:13]:
```suggestion
include 'repository/config/os';
```

File: clusters/grid/umd4/repository/config.pan [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+# Local Repositories
+variable YUM_SITE_SNAPSHOT_NS ?= YUM_SNAPSHOT_NS;
+variable SITE_REPOSITORY_LIST ?= list();
+variable SITE_REPOSITORY_CONFIG ?= dict();
+variable DEBUG = debug(
+    'OS_REPOSITORY_LIST = ' + to_string(OS_REPOSITORY_LIST) + "\n" +
+    'SITE_REPOSITORY_LIST = ' + to_string(SITE_REPOSITORY_LIST) + "\n" +
+    'SITE_REPOSITORY_CONFIG = ' + to_string(SITE_REPOSITORY_CONFIG) + "\n"
+);
+include'quattor/functions/repository';
```
Comment (jrha) [inline clusters/grid/umd4/repository/config.pan:27]:
```suggestion
include 'quattor/functions/repository';
```

**Comment by jouvin** [comment]
@jrha it should now be in a good shape! The only remaining problem is due to the missing `quattor/aii/config.pan` in `template-library-core` due to the releaser issues identified during 24.10.0-rc3 (https://github.com/quattor/release/issues/360). I'd suggest to add temporarily (and manually) an empty version of `quattor/aii/config.pan` : I checked that it is enough to fix the problem in `test-library` and may limit the chicken&egg problem that we have...

**Comment by jrha** [comment]
Restore AII templates: https://github.com/quattor/template-library-core/pull/234

**Comment by jouvin** [comment]
All green now!

**Comment by jrha** [comment]
Excellent, thanks for the work on this!

**Comment by jouvin** [comment]
I suggest to wait a little bit to merge this one. I'm adding in `get-template-library` the code required to support natively GitHub actions rather than defining the TRAVIS variables in `test-library`. Should be quick and will fix the problem in all repositories.

**Comment by jouvin** [comment]
Works as expected with the new `get-template-library`. Can be merged.

**Review by jrha — APPROVED** [review]

**Pull request merged**